core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -699,12 +853,24 @@ private class FsApplicationAttemptInfo(
     endTime: Long,
     lastUpdated: Long,
     sparkUser: String,
-    completed: Boolean = true)
-  extends ApplicationAttemptInfo(
-      attemptId, startTime, endTime, lastUpdated, sparkUser, completed)
+    completed: Boolean = true,
+    val fileSize: Long = -1,
+    val instance: Long = -1)
```

Comment:
modtime was proving too brittle. Not all updates were altering the modification time; I wasted a lot of time tying to de-brittalise tests to the point where I felt that while the test was working, I couldn't trust it in production. And I say that as the person who formally specified the Hadoop FS behaviours.